scen_edit/command/save_command.lua

```diff
@@ -122,36 +122,43 @@ function SaveCommand.GenerateScript(dev)
                 TeamLeader = 0,
                 AllyTeam = team.allyTeam,
                 RGBColor = team.color.r .. " " .. team.color.g .. " " .. team.color.b,
+                Side = team.side,
             })
-        end
-        if team.ai then
-            local aiShortName = "NullAI"
-            local aiVersion = ""
-            if not dev then
-                -- TODO: Support other AIs for non-dev scripts
-            end
+            if team.ai then
+                local aiShortName = "NullAI"
+                local aiVersion = ""
+                if not dev then
+                    -- TODO: Support other AIs for non-dev scripts
+                end
 
-            table.insert(ais, {
-                Name = team.name,
-                Team = team.id - 1,
-                ShortName = aiShortName,
-                Version = aiVersion,
+                table.insert(ais, {
+                    Name = team.name,
+                    -- Teams are referenced by the index, which may match with
+                    -- team.id, or may not, depending on the order they are
+                    -- parsed.
+                    Team = #teams - 1,
```

Comment:
I'm not sure original code is correct, but if we don't use proper team IDs, it's quite likely scenario logic could fail as it depends on fixed team IDs.
Generally don't like using the #teams counter. Can this be implemented with just team.id, but correctly?